File: guides/voice/respond-incoming-calls-guide/respond-with-parameters/example.5.x.cs
// In Package Manager, run:
// Install-Package Twilio.Mvc -Pre

using System.Web.Mvc;
using Twilio.TwiML;

public class VoiceController : Controller
{
    [HttpPost]
    public ActionResult Index(string fromCity)
    {
        // Get information about the incoming call, like the city associated
        // with the phone number (if Twilio can discover it)
        var city = fromCity;

        var response = new VoiceResponse();
        response.Say($"Never gonna give you up {city}", voice: "alice");
        response.Play("https://demo.twilio.com/docs/classic.mp3");

        return Content(response.ToString(), "text/xml")
    }
}

File: guides/voice/respond-incoming-calls-guide/respond-with-parameters/example.5.x.rb
# Get twilio-ruby from twilio.com/docs/ruby/install
require 'rubygems' # This line not needed for ruby > 1.8
require 'sinatra'
require 'twilio-ruby'

post '/voice' do
  city = params['FromCity']

  # Start our TwiML response
  Twilio::TwiML::VoiceResponse.new do |r|
    # Use <Say> to give the caller some instructions
    r.say("Never gonna give you up #{city}.", voice: 'alice')
    r.play(url: 'https://demo.twilio.com/docs/classic.mp3')
  end.to_s
end
